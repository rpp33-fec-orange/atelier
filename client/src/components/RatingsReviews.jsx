import React from 'react';

const RatingsReviews = (props) => (
  <div>
    <h4> Ratings and Reviews </h4>
  </div>
)

export default RatingsReviews;