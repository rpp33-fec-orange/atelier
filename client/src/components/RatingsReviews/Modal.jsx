import React from 'react';

class Modal extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			productName: ''
		}
	}

	componentDidMount() {
		this.setState({
			productName: this.props.productName
		});
	}

	componenetDidUpdate(prevProps) {
		if (this.props.productName !== prevProps.productName) {
			var productName = this.props.productName;
			this.setState({
				productName: productName
			});
		}
	}

	render() {
		const showHideClassName = this.props.show ? "modal-rr display-block" : "modal-rr display-none";

		return (
			<div className={showHideClassName} >
				<section className="modal-rr-main">
					<div className="modal-rr-container">
						<h1>Write Your Review</h1>
						<p>About the {this.props.productName}: </p>
						<div className="modal-rr-overallrating">
							&#9734;&#9734;&#9734;&#9734;&#9734;
						</div>
						<p>Do you recommend this product?</p>
						<div className="modal-rr-recommended" >
							<div>
								<input type="radio" id="yes" name="recommended" value="yes" onChange={this.props.writeRecommended} />
								<label htmlFor="yes">Yes</label>
							</div>
							<div>
								<input type="radio" id="no" name="recommended" value="no" onChange={this.props.writeRecommended} />
								<label htmlFor="no">No</label>
							</div>
						</div>
						<br></br>
						<div className="characteristics">
							Characteristics:
							<div>Size:
								<input type="radio" id="size1" name="characteristics-size" value="size1" onChange={this.props.writeSize} />
								<label htmlFor="size1">1</label>
								<input type="radio" id="size2" name="characteristics-size" value="size2" onChange={this.props.writeSize} />
								<label htmlFor="size2">2</label>
								<input type="radio" id="size3" name="characteristics-size" value="size3" onChange={this.props.writeSize} />
								<label htmlFor="size3">3</label>
								<input type="radio" id="size4" name="characteristics-size" value="size4" onChange={this.props.writeSize} />
								<label htmlFor="size4">4</label>
								<input type="radio" id="size5" name="characteristics-size" value="size5" onChange={this.props.writeSize} />
								<label htmlFor="size5">5</label>
							</div>
							<div>Width:
								<input type="radio" id="width1" name="characteristics-width" value="width1" onChange={this.props.writeWidth} />
								<label htmlFor="width1">1</label>
								<input type="radio" id="width2" name="characteristics-width" value="width2" onChange={this.props.writeWidth} />
								<label htmlFor="width2">2</label>
								<input type="radio" id="width3" name="characteristics-width" value="width3" onChange={this.props.writeWidth} />
								<label htmlFor="width3">3</label>
								<input type="radio" id="width4" name="characteristics-width" value="width4" onChange={this.props.writeWidth} />
								<label htmlFor="width4">4</label>
								<input type="radio" id="width5" name="characteristics-width" value="width5" onChange={this.props.writeWidth} />
								<label htmlFor="width5">5</label>
							</div>
							<div>Comfort:
								<input type="radio" id="comfort1" name="characteristics-comfort" value="comfort1" onChange={this.props.writeComfort} />
								<label htmlFor="comfort1">1</label>
								<input type="radio" id="comfort2" name="characteristics-comfort" value="comfort2" onChange={this.props.writeComfort} />
								<label htmlFor="comfort2">2</label>
								<input type="radio" id="comfort3" name="characteristics-comfort" value="comfort3" onChange={this.props.writeComfort} />
								<label htmlFor="comfort3">3</label>
								<input type="radio" id="comfort4" name="characteristics-comfort" value="comfort4" onChange={this.props.writeComfort} />
								<label htmlFor="comfort4">4</label>
								<input type="radio" id="comfort5" name="characteristics-comfort" value="comfort5" onChange={this.props.writeComfort} />
								<label htmlFor="comfort5">5</label>
							</div>
							<div>Quality:
								<input type="radio" id="quality1" name="characteristics-quality" value="quality1" onChange={this.props.writeQuality} />
								<label htmlFor="comfort1">1</label>
								<input type="radio" id="quality2" name="characteristics-quality" value="quality2" onChange={this.props.writeQuality} />
								<label htmlFor="comfort2">2</label>
								<input type="radio" id="quality3" name="characteristics-quality" value="quality3" onChange={this.props.writeQuality} />
								<label htmlFor="comfort3">3</label>
								<input type="radio" id="quality4" name="characteristics-quality" value="quality4" onChange={this.props.writeQuality} />
								<label htmlFor="comfort4">4</label>
								<input type="radio" id="quality5" name="characteristics-quality" value="quality5" onChange={this.props.writeQuality} />
								<label htmlFor="comfort5">5</label>
							</div>
							<div>Length:
								<input type="radio" id="length1" name="characteristics-length" value="length1" onChange={this.props.writeLength} />
								<label htmlFor="length1">1</label>
								<input type="radio" id="length2" name="characteristics-length" value="length2" onChange={this.props.writeLength} />
								<label htmlFor="length2">2</label>
								<input type="radio" id="length3" name="characteristics-length" value="length3" onChange={this.props.writeLength} />
								<label htmlFor="length3">3</label>
								<input type="radio" id="length4" name="characteristics-length" value="length4" onChange={this.props.writeLength} />
								<label htmlFor="length4">4</label>
								<input type="radio" id="length5" name="characteristics-length" value="length5" onChange={this.props.writeLength} />
								<label htmlFor="length5">5</label>
							</div>
							<div>Fit:
								<input type="radio" id="fit1" name="characteristics-fit" value="fit1" onChange={this.props.writeFit}/>
								<label htmlFor="fit1">1</label>
								<input type="radio" id="fit2" name="characteristics-fit" value="fit2" onChange={this.props.writeFit}/>
								<label htmlFor="fit2">2</label>
								<input type="radio" id="fit3" name="characteristics-fit" value="fit3" onChange={this.props.writeFit}/>
								<label htmlFor="fit3">3</label>
								<input type="radio" id="fit4" name="characteristics-fit" value="fit4" onChange={this.props.writeFit}/>
								<label htmlFor="fit4">4</label>
								<input type="radio" id="fit5" name="characteristics-fit" value="fit5" onChange={this.props.writeFit}/>
								<label htmlFor="fit5">5</label>
							</div>
						</div>
						<br></br>
						<div className="review-summary">
							<label htmlFor="summary">Review summary:    </label>
							<input type="text" id="summary" name="summary" required
								minLength="4" maxLength="60" size="70" onChange={this.props.writeReviewSummary}/>
						</div>
						<br></br>
						<div className="review-body">
							<label htmlFor="body">Review body:    </label>
							<textarea id="body" name="body" rows="5" cols="33" defaultValue="Why did you like the product or not?" required
								minLength="50" maxLength="1000" size="1020" onChange={this.props.writeReviewBody}>
							</textarea>
						</div>
						<br></br>
						<div className="upload-photo">Upload your photos:
							<button className="upload" type="button" onChange={this.props.writeUploadPhotos}>
								Upload
							</button>
						</div>
						<br></br>
						<div className="nickname">
							<label htmlFor="nickname">What is your nickname?    </label>
							<input type="text" id="nickname" name="nickname" placeholder="Example: jackson11!" required
								minLength="1" maxLength="60" size="70" onChange={this.props.writeNickname}/>
							<sub>   For privacy reasons, do not use your full name or email address</sub>
						</div>
						<br></br>
						<div className="email">
							<label htmlFor="email">Your email:    </label>
							<input type="text" id="email" name="email" placeholder="Example: jackson11@email.com" required
								minLength="1" maxLength="60" size="70" onChange={this.props.writeEmail}/>
							<sub>    For authentication reasons, you will not be emailed</sub>
						</div>
						<br></br>
						<div>
							<button className="submit" type="button">
								Submit
							</button>
						</div>
						<button className="close" type="button" onClick={this.props.handleClose}>
							Close
						</button>
					</div>
				</section>
			</div>
		);
	}
}


export default Modal;


