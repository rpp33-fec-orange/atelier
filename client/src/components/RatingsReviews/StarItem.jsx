import React from 'react';

class StarItem extends React.Component {
  constructor(props) {
    super(props);
    this.roundToQuater = this.roundToQuater.bind(this);
  }
  roundToQuater(n) {
    var number = n;
    if (number >= 0 && number <= 5) {
      number = (Math.round(number * 4) / 4).toFixed(2);
      return number;
    }
    return ('number has to be between 0 and 5');
  }

  render() {
    const starsInner = {
      position: 'absolute',
      top: 0,
      left: 0,
      overflow: 'hidden',
      width: 50,
    };

    const starsOuter = {
      display: 'inline-block',
      position: 'relative',
      fontFamily: 'FontAwesome',
    }

    return (
      <div></div>
    )
  }
}

export default StarItem;



