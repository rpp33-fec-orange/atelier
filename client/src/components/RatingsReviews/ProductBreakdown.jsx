import React from 'react';
import RatingsReviews from './RatingsReviews.jsx';

class ProductBreakdown extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			size: null,
			width: null,
			comfort: null,
			quality: null,
			length: null,
			fit: null,
			percentSize: null,
			percentWidth: null,
			percentComfort: null,
			percentQuality: null,
			percentLength: null,
			percentFit: null

		};
	}

	breakDown() {
		var size = null;
		var width = null;
		var comfort = null;
		var quality = null;
		var length = null;
		var fit = null;
		var percentSize = null;
		var percentWidth = null;
		var percentComfort = null;
		var percentQuality = null;
		var percentLength = null;
		var percentFit = null;

		if (this.props.meta_characteristics.Size != undefined) {
			if (this.props.meta_characteristics.Width.value != undefined) {
				size = (this.props.meta_characteristics.Size.value === undefined) ? '0' : this.props.meta_characteristics.Size.value;
				percentSize = Math.round((size / 5) * 100);
			}
		}
		if (this.props.meta_characteristics.Width != undefined) {
			if (this.props.meta_characteristics.Width.value != undefined) {
				width = (this.props.meta_characteristics.Width.value === undefined) ? '0' : this.props.meta_characteristics.Width.value;
				percentWidth = Math.round((width / 5) * 100);
			}

		}
		if (this.props.meta_characteristics.Comfort != undefined) {
			if (this.props.meta_characteristics.Comfort.value != undefined) {
				comfort = (this.props.meta_characteristics.Comfort.value === undefined) ? '0' : this.props.meta_characteristics.Comfort.value;
				percentComfort = Math.round((comfort / 5) * 100);
			}
		}
		if (this.props.meta_characteristics.Quality != undefined) {
			if (this.props.meta_characteristics.Quality.value != undefined) {
				quality = (this.props.meta_characteristics.Quality.value === undefined) ? '0' : this.props.meta_characteristics.Quality.value;
				percentQuality = Math.round((quality / 5) * 100);
			}
		}
		if (this.props.meta_characteristics.Length != undefined) {
			if (this.props.meta_characteristics.Length.value != undefined) {
				length = (this.props.meta_characteristics.Length.value === undefined) ? '0' : this.props.meta_characteristics.Length.value;
				percentLength = Math.round((length / 5) * 100);
			}
		}
		if (this.props.meta_characteristics.Fit != undefined) {
			if (this.props.meta_characteristics.Fit.value != undefined) {
				fit = (this.props.meta_characteristics.Fit.value === undefined) ? '0' : this.props.meta_characteristics.Fit.value;
				percentFit = Math.round((fit / 5) * 100);
			}
		}
		this.setState({
			size: size,
			width: width,
			comfort: comfort,
			quality: quality,
			length: length,
			fit: fit,
			percentSize: percentSize,
			percentWidth: percentWidth,
			percentComfort: percentComfort,
			percentQuality: percentQuality,
			percentLength: percentLength,
			percentFit: percentFit
		})
	}

	componentDidMount() {
		this.breakDown();
	}

	componentDidUpdate(prevProps) {
		if (this.props.meta_characteristics != prevProps.meta_characteristics) {
			this.breakDown();
		}
	}

	render() {
		let sizeRender;
		let widthRender;
		let comfortRender;
		let qualityRender;
		let lengthRender;
		let fitRender;

		if (this.state.size != null) {
			sizeRender =
				<div>Size
					<div className="size-container">
						<div className="size-side"></div>
						<div className="size-marker" style={{ marginLeft: `${this.state.percentSize}%` }}>&#9660;</div>
					</div>
					<div className="size-explanation">
						<div className="side">Too Small</div>
						<div className="middle">Perfect</div>
						<div className="side">Too Wide</div>
					</div>
				</div>
		}
		if (this.state.width != null) {
			widthRender =
				<div>Width
					<div className="width-container">
						<div className="width-side"></div>
						<div className="width-marker" style={{ marginLeft: `${this.state.percentWidth}%` }}>&#9660;</div>
					</div>
					<div className="width-explanation">
						<div className="side">Too Narrow</div>
						<div className="middle">Perfect</div>
						<div className="side">Too Wide</div>
					</div>
				</div>
		}
		if (this.state.comfort != null) {
			comfortRender =
				<div>Comfort
					<div className="comfort-container">
						<div className="comfort-side"></div>
						<div className="comfort-marker" style={{ marginLeft: `${this.state.percentComfort}%` }}>&#9660;</div>
					</div>
					<div className="comfort-explanation">
						<div className="side">Uncomfortable</div>
						<div className="middle">Ok</div>
						<div className="side">Perfect</div>
					</div>
				</div>
		}
		if (this.state.quality != null) {
			qualityRender =
				<div>Quality
					<div className="quality-container">
						<div className="quality-side"></div>
						<div className="quality-marker" style={{ marginLeft: `${this.state.percentQuality}%` }}>&#9660;</div>
					</div>
					<div className="quality-explanation">
						<div className="side">Poor</div>
						<div className="middle">What I Expected</div>
						<div className="side">Perfect</div>
					</div>
				</div>
		}
		if (this.state.length != null) {
			lengthRender =
				<div>Length
					<div className="length-container">
						<div className="length-side"></div>
						<div className="length-marker" style={{ marginLeft: `${this.state.percentLength}%` }}>&#9660;</div>
					</div>
					<div className="length-explanation">
						<div className="side">Runs Short</div>
						<div className="middle">Perfect</div>
						<div className="side">Runs Long</div>
					</div>
				</div>
		}
		if (this.state.fit != null) {
			fitRender =
				<div>Fit
					<div className="fit-container">
						<div className="fit-side"></div>
						<div className="fit-marker" style={{ marginLeft: `${this.state.percentFit}%` }}>&#9660;</div>
					</div>
					<div className="fit-explanation">
						<div className="side">Runs Tight</div>
						<div className="middle">Perfect</div>
						<div className="side">Runs Long</div>
					</div>
				</div>
		}
		return (
			<div>
				{sizeRender}
				{widthRender}
				{comfortRender}
				{qualityRender}
				{lengthRender}
				{fitRender}
			</div>
		);
	}
}

export default ProductBreakdown;