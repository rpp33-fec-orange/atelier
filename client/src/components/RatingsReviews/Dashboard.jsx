import React from 'react';
import Modal from './Modal.js';

class Dashboard extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
      show: false
		};
		this.showModal = this.showModal.bind(this);
		this.hideModal = this.hideModal.bind(this);
	}

	showModal = () => {
    this.setState({ show: true });
	};

	hideModal = () => {
    this.setState({ show: false });
	};

	render() {
		return (
			<div>
        <button id="modal-btn" onClick={this.showModal} >This is a Modal</button>
			  <div className="modal">Hello
					<div className="modal-content">
          <span className="close-btn" onClick={this.hideModal}>&times;</span>
					<p>this is the text inside the modal</p>
				  </div>
				</div>
			</div>
		);
	}
}

export default Dashboard;