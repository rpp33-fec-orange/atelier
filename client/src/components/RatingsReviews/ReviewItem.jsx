import React from 'react';
import Modal from './Modal.jsx';
import Dashboard from './Dashboard.jsx';
import StarRating from './StarRating.jsx';

class ReviewItem extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			review: {},
			reviewID: 0,
			recommend: true,
			ready: false,
			ratingReview: 0,
			helpfulness: 0,
			markedHelpful: false,
			reported: false
		}
		this.getRecommendation = this.getRecommendation.bind(this);
		this.markReviewHelpful = this.markReviewHelpful.bind(this);
		this.reportReview = this.reportReview.bind(this);
	}

	getRecommendation(event) {
		event.preventDefault();
		this.setState({
			recommend: event.target.value
		});
	}

	isReady() {
		var propsReady = this.state.ready;

	}

	markReviewHelpful() {

	}

	reportReview() {

	}

	componentDidMount() {
		this.setState({
			review: this.props.review,
			ratingReview: this.props.review.rating,
			helpfulness: this.props.review.helpfulness,
			reviewID: this.props.review - id
		});
	}


	render() {
		console.log('this is this.state.review: ', this.state.review);
		return (
			<div className="wrapper">
				<div className="one" id="review-list">
					<div className="star-review-container">
						<StarRating num={this.props.review.rating} />
						<div className="user-and-date">
							{this.state.review.reviewer_name}, {this.state.review.date}
						</div>
					</div>
					<h3 className="review-summary">{this.state.review.summary}</h3>
					<p className="review-body">{this.state.review.body}</p>
					<div className="helpful">Helpful?
						{
							!this.state.markedHelpful
								?
								<span className="helpful-unmarked" onClick={this.markReviewHelpful}>Yes</span>
								:
								<span className="helpful-marked">Yes</span>
						}
						{
							`${this.state.helpfulness}   |   `
						}
						{
							!this.state.reported
							?
							<span className="report-unmarked" onClick={this.reportReview}>Report</span>
							:
							<span className="report-marked">Report</span>
						}
					</div>
					{this.state.review.recommend &&
						<div>
							âœ” I recommend this product
						</div>
					}
					<div className="productPhoto">
						<sub>Product Photo: </sub>
						{this.state.review.photos.length > 0
							? <div>
								{this.state.review.photos}
							</div>
							: <div>
								<img src='https://i.vimeocdn.com/portrait/1274237_300x300.jpg' />
							</div>
						}
					</div>
					<hr></hr>
				</div>
			</div>
		);
	}
}

export default ReviewItem;