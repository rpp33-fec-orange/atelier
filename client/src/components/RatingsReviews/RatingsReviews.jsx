import React from 'react';
import $ from 'jquery';
import ReviewItem from './ReviewItem.jsx';
import StarNumber from './StarNumber.jsx';
import StarList from './StarList.jsx';
import StarItem from './StarItem.jsx';
import ComfortSlider from './ComfortSlider.jsx';
import SizeSlider from './SizeSlider.jsx';

class RatingsReviews extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      reviews: []
    }
    this.reviewHandler = this.reviewHandler.bind(this);
  }

  reviewHandler() {
    var url = '/'
   $.ajax({
     method: "GET",
     url: url,
     success: (reviews) => {
      this.setState({
        reviews: reviews
      });
     },
     error: (err) => {
      console.log('error from get request');
     }
   })
   .done(function() {
    console.log('get request is done');
   });
  }

  render() {
    return (
      <div>
        <h1 id='ratings-reviews'>RATINGS AND REVIEWS</h1>
        <StarNumber />
        <StarList />
        <SizeSlider />
        <ComfortSlider />
        <h3>248 reviews, sorted by relevance</h3>
        {/* <div> Review #1 </div>
      <div> Review #2 </div> */}
        <ReviewItem />
      </div>
    );
  }
}

export default RatingsReviews;