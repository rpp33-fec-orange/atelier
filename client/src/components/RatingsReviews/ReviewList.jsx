import React from 'react';
import ReviewItem from './ReviewItem.jsx';
import Dashboard from './Dashboard.jsx';

class ReviewList extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			reviewList: [],
			renderedReviews: [],
			holdingReviews: [],
			reviewsPerClick: 2,
			currentIndex: 0,
			showMore: 'more-reviews-btn'
		}
		this.mappingReviews = this.mappingReviews.bind(this);
		this.handleShowMoreReviews = this.handleShowMoreReviews.bind(this);
		this.slicedReviewList = this.slicedReviewList.bind(this);
		this.setShowMore = this.setShowMore.bind(this);
	}

	componentDidMount() {
		// console.log('finally, this.props.reviews is: ', this.props.reviews);
		var reviewList = this.props.reviews;
		this.setState({
			reviewList: reviewList,
		}, this.slicedReviewList);
		// console.log('finally, this.state.reviewList is: ', this.state.reviewList);
	}

	componentDidUpdate(prevProps) {
		if (this.props.reviews !== prevProps.reviews) {
			var reviewList = this.props.reviews;
			this.setState({
				reviewList: reviewList,
			}, this.slicedReviewList);
		}
	}

	mappingReviews() {
		// console.log('props.reviews in Reviewlist: ', this.props.reviews);
		var reviews = this.state.holdingReviews;
		var renderedReviews = reviews.map((review) =>
			<ReviewItem key={review.review_id.toString()} review={review} onMarkedHelpful={this.props.onMarkedHelpful} onMarkedReported={this.props.onMarkedReported} />
		);
		this.setState({
			renderedReviews: renderedReviews
		}, this.setShowMore);
	}

	setShowMore() {
		var index = this.state.reviewsPerClick;
		var length = this.state.reviewList.length;
		if (index >= length) {
			this.setState({
				showMore: 'more-reviews-btn-hide'
			});
		}
	}

	slicedReviewList() {
		var currentIndex = this.state.currentIndex;
		var reviewsPerClick = this.state.reviewsPerClick;
		const slicedReviewList = this.state.reviewList.slice(currentIndex, reviewsPerClick);
		this.setState({
			holdingReviews: slicedReviewList
		}, this.mappingReviews);

	}

	handleShowMoreReviews() {
		// console.log('write logic for showing more reviews here!');
		var currentReviewsPerClick = this.state.reviewsPerClick;
		currentReviewsPerClick += 2;
		this.setState({
			reviewsPerClick: currentReviewsPerClick
		}, this.slicedReviewList);
	}


	// needs an array to store initally only 2 reviews, then add 2 more each time the Add More button is clicked
	// When no more reviews can be added, such as when length of this.state.reviews is reached, hide the Add More button.

	render() {
		var showMore = this.state.showMore;
		return (
			<div id="review-list-container">
				<div className="scroller">
					{this.state.renderedReviews}
				</div>
				<div className="buttons" id="buttons">
					<button className={showMore} onClick={this.handleShowMoreReviews}>MORE REVIEWS</button>
					<Dashboard />
				</div>
			</div>
		)
	}
}

export default ReviewList;