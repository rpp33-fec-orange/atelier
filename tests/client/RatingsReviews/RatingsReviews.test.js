import React from 'react';
import TestRenderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import { configure } from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow } from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });
import '@testing-library/jest-dom'
import RatingsReviews from '../../../client/src/components/RatingsReviews/RatingsReviews.jsx';
import Dashboard from '../../../client/src/components/RatingsReviews/Dashboard.jsx';
import Modal from '../../../client/src/components/RatingsReviews/Modal.jsx';
import ProductBreakdown from '../../../client/src/components/RatingsReviews/ProductBreakdown.jsx';
import ReviewItem from '../../../client/src/components/RatingsReviews/ReviewItem.jsx';
import ReviewList from '../../../client/src/components/RatingsReviews/ReviewList.jsx';
import StarItem from '../../../client/src/components/RatingsReviews/StarItem.jsx';
import StarList from '../../../client/src/components/RatingsReviews/StarList.jsx';
import StarNumber from '../../../client/src/components/RatingsReviews/StarNumber.jsx';
import StarRating from '../../../client/src/components/RatingsReviews/StarRating.jsx';







describe('RatingsReviews component tests', () => {
	const testRenderer = TestRenderer.create(<RatingsReviews />);
	const testInstance = testRenderer.root;

	test('it should contain an h3 element with the text "RATINGS AND REVIEWS', () => {
		expect(testInstance.findByProps("div").children).toEqual(['RATINGS AND REVIEWS']);
	})

	// it("renders RatingsReviews header", () => {
	// 	const wrapper = shallow(<RatingsReviews />);
	// 	const ratingsReviews = <h3>RATINGS AND REVIEWS</h3>
	// 	expect(wrapper.contains(ratingsReviews)).toEqual(true);
	// })

	test('Snapshot should match!', () => {
		let tree = testRenderer.toJSON();
		expect(tree).toMatchSnapshot();
	})

	it('it should render StarNumber.jsx component', () => {
		const component = shallow(<StarNumber />);
		expect(component.exists()).toEqual(true);
	});

	it('it should render StarList.jsx component', () => {
		const component = shallow(<StarList />);
		expect(component.exists()).toEqual(true);
	});

	it('it should render StarItem.jsx component', () => {
		const component = shallow(<StarItem />);
		expect(component.exists()).toEqual(true);
	});

	it('it should render StarRating.jsx component', () => {
		const component = shallow(<StarRating />);
		expect(component.exists()).toEqual(true);
	});

	it('it should render ReviewList.jsx component', () => {
		const component = shallow(<ReviewList />);
		expect(component.exists()).toEqual(true);
	});

	it('it should render ReviewItem.jsx component', () => {
		const component = shallow(<ReviewItem />);
		expect(component.exists()).toEqual(true);
	});

	it('it should render ProductBreakdown.jsx component', () => {
		const component = shallow(<ProductBreakdown />);
		expect(component.exists()).toEqual(true);
	})
})

