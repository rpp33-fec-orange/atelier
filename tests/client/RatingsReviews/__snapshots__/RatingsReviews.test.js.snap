// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingsReviews component tests Snapshot should match! 1`] = `
<div>
  Loading Ratings and Reviews...
</div>
`;
